docker/pulumi/Dockerfile

```diff
@@ -94,6 +105,23 @@ RUN curl -L https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-
   helm repo add stable https://charts.helm.sh/stable && \
   helm repo update
 
+# Python
+# Install Pyenv and preinstall supported Python versions
+RUN git clone --depth=1 https://github.com/pyenv/pyenv.git /pyenv
+ENV PYENV_ROOT /pyenv
+ENV PATH="${PYENV_ROOT}/shims:${PYENV_ROOT}/bin:${PATH}"
+RUN pyenv install 3.12
+RUN pyenv install 3.11
+RUN pyenv install 3.10
+RUN pyenv install 3.9
+RUN pyenv global 3.9 # Default version
+# Poetry
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH="/root/.local/bin:${PATH}"
```

Comment:
Update this to /usr/local/share/poetry, see https://github.com/pulumi/pulumi-docker-containers/pull/240/files#diff-56dcbb4f608b8a8d5169091d1a21b391651622f6404e5ef885198868f23a2984R30